x-pack/plugins/infra/server/lib/snapshot/types.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { JsonObject } from '../../../common/typed_json';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { InfraSourceConfiguration } from '../../../public/graphql/types';
```

Comment:
This should also exist in `../../../common/graphql/types`, which can be imported without an eslint override.